lib/genesis_collector/megaraid.rb

```diff
@@ -0,0 +1,46 @@
+module GenesisCollector
+  module MegaRaid
+
+    def collect_megaraid
+      @payload[:raids] ||= []
+      @payload[:properties] ||= {}
+      unless read_megacli_adapter_type.nil?
+        @payload[:properties]['MEGARAID_TYPE'] = read_megacli_adapter_type
+        @payload[:properties]['MEGARAID_FW_PACKAGE'] = read_megacli_adapter_firmware
+        @payload[:raids].concat read_megacli_logical_disks
+      end
+    end
+
+    private
+
+    def read_megacli_adapter_type()
+      @megacli_adapter_output ||= shellout_with_timeout('megacli -AdpAllInfo -aAll', 10)
+      match = @megacli_adapter_output.match(/Product Name\s*:\s*(.+)\s*$/)
+      return nil if match.nil?
+      match[1]
+    end
+
+    def read_megacli_adapter_firmware()
+      @megacli_adapter_output ||= shellout_with_timeout('megacli -AdpAllInfo -aAll', 10)
+      match = @megacli_adapter_output.match(/FW Package Build\s*:\s*(\S+)$/)
+      return nil if match.nil?
+      match[1]
+    end
+
+    def read_megacli_logical_disks()
+      @megacli_ld_output ||= shellout_with_timeout('megacli -LDInfo -Lall -aAll', 10)
```

Comment:
to cache output already collected .. might not be necessary, as only run once.

not datastores app collecting this data (yet), but we've had surprises about LDs on backup hosts. adding a genesis visibility module might be the best path to provide visibility.